samples/Monitoring/deploy-mmaextension-windows-tags-loganalytics/azurepolicy.json

```diff
@@ -0,0 +1,140 @@
+{
+    "properties": {
+        "displayName": "Deploy Microsoft Monitoring Agent on Windows Machines based on a Tag",
+        "mode": "Indexed",
+        "description": "Deploy Microsoft Monitoring Agent on Windows Machines based on a Tag with value: MMA and Name: Windows",
+        "metadata": {
+            "category": "Batch"
+        },
+        "parameters": {
+            "logAnalytics": {
+                "type": "String",
+                "metadata": {
+                    "displayName": "Log Analytics workspace",
+                    "description": "Select Log Analytics workspace from dropdown list. If this workspace is outside of the scope of the assignment you must manually grant 'Log Analytics Contributor' permissions (or similar) to the policy assignment's principal ID.",
+                    "strongType": "omsWorkspace",
+                    "assignPermissions": true
+                }
+            },
+            "tagName": {
+                "type": "String",
+                "metadata": {
+                    "displayName": "Tag Name",
+                    "description": "Name of the tag, such as 'production'"
+                }
+            },
+            "WorkspaceID": {
+                "Type": "String",
+                "metadata": {
+                    "displayName": "Log Analytics workspace",
+                    "description": "Name of the workspace, such as 'production'"
+                }
+            }
+        },
+        "policyRule": {
+            "if": {
+                "allOf": [{
+                        "field": "type",
+                        "equals": "Microsoft.Compute/virtualMachines"
+                    },
+                    {
+                        "field": "[concat('tags[', parameters('tagName'), ']')]",
+                        "equals": "Windows"
+                    }
+                ]
+            },
+            "then": {
+                "effect": "deployIfNotExists",
+                "details": {
+                    "type": "Microsoft.Compute/virtualMachines/extensions",
+                    "roleDefinitionIds": [
+                        "/providers/microsoft.authorization/roleDefinitions/92aaf0da-9dab-42b6-94a3-d43ce8d16293"
+                    ],
+                    "existenceCondition": {
+                        "allOf": [{
+                                "field": "Microsoft.Compute/virtualMachines/extensions/type",
+                                "equals": "MicrosoftMonitoringAgent"
+                            },
+                            {
+                                "field": "Microsoft.Compute/virtualMachines/extensions/publisher",
+                                "equals": "Microsoft.EnterpriseCloud.Monitoring"
+                            },
+                            {
+                                "field": "Microsoft.Compute/virtualMachines/extensions/provisioningState",
+                                "equals": "Succeeded"
+                            }
```

Comment:
I know the built-in does this but this typically isn't a good idea.  If a user-provided LA extension isn't done deploying or is in a bad state this will add a new LA extension with the name specified in this policy.  The policy should only add an LA extension if one does not exist.  If you want to audit failed LA extensions that should be its own policy

Reply:
I understand this and but since the built-in does this hence I went with the same approach. I will come up with an Audit policy as well. Thanks